WordPress/Classes/ViewRelated/Blog/Blog Details/BlogDetailsViewController+Swift.swift

```diff
@@ -0,0 +1,326 @@
+        let menusVC = MenusViewController.withJPBannerForBlog(blog)
+        presentationDelegate?.presentBlogDetailsViewController(menusVC)
+    }
+
+    @objc(showCommentsFromSource:)
+    func showComments(from source: BlogDetailsNavigationSource) {
+        trackEvent(.openedComments, from: source)
+
+        guard let commentsVC = CommentsViewController(blog: blog) else {
+            return wpAssertionFailure("failed to instantiate")
+        }
+        commentsVC.navigationItem.largeTitleDisplayMode = .never
+
+        if isSidebarModeEnabled {
+            commentsVC.isSidebarModeEnabled = true
+
+            let splitVC = UISplitViewController(style: .doubleColumn)
+            splitVC.presentsWithGesture = false
+            splitVC.preferredDisplayMode = .oneBesideSecondary
+            splitVC.preferredPrimaryColumnWidth = 320
+            splitVC.minimumPrimaryColumnWidth = 375
+            splitVC.maximumPrimaryColumnWidth = 400
+            splitVC.setViewController(commentsVC, for: .primary)
+
+            let noSelectionVC = UIViewController()
+            noSelectionVC.view.backgroundColor = .systemBackground
+            splitVC.setViewController(noSelectionVC, for: .secondary)
+
+            presentationDelegate?.presentBlogDetailsViewController(splitVC)
+        } else {
+            presentationDelegate?.presentBlogDetailsViewController(commentsVC)
+        }
+    }
+
+    @objc func showPlugins() {
+        WPAppAnalytics.track(.openedPluginDirectory, blog: blog)
+
+        if Feature.enabled(.pluginManagementOverhaul) {
+            showManagePluginsScreen()
+            return
+        }
+
+        guard let site = JetpackSiteRef(blog: blog) else {
+            return wpAssertionFailure("unexpected blog")
+        }
+        let controller = PluginDirectoryViewController(site: site)
+        controller.navigationItem.largeTitleDisplayMode = .never
+        presentationDelegate?.presentBlogDetailsViewController(controller)
+    }
+
+    @objc(showStatsFromSource:)
+    func showStats(from source: BlogDetailsNavigationSource) {
+        trackEvent(.statsAccessed, from: source)
+
+        let statsVC = makeStatsVC()
+
+        // Calling `showDetailViewController:sender:` should do this automatically for us,
+        // but when showing stats from our 3D Touch shortcut iOS sometimes incorrectly
+        // presents the stats view controller as modal instead of pushing it. As a
+        // workaround for now, we'll manually decide whether to push or use `showDetail`.
+        // @frosty 2016-09-05
+        if let splitViewController, splitViewController.isCollapsed {
+            navigationController?.pushViewController(statsVC, animated: true)
+        } else {
+            presentationDelegate?.presentBlogDetailsViewController(statsVC)
+        }
+    }
+
+    private func makeStatsVC() -> UIViewController {
+        guard JetpackFeaturesRemovalCoordinator.jetpackFeaturesEnabled() else {
```

Comment:
Would it make sense to use 

```suggestion
        guard BuildSettings.current.brand == .jetpack else {
```

Just to progressively move away from that migration coordinator that I think is no longer relevant?

Reply:
I'd keep it for now. I'm not 100% certain what we are going to do with the migration code.